crates/proof-impl/btc-blockspace/src/logic.rs

```diff
@@ -2,59 +2,64 @@
 
 use bitcoin::Block;
 use borsh::{BorshDeserialize, BorshSerialize};
-use strata_primitives::{buf::Buf32, params::RollupParams};
+use strata_primitives::{block_credential::CredRule, buf::Buf32};
 use strata_state::{batch::BatchCheckpoint, tx::DepositInfo};
+use strata_tx_parser::filter::TxFilterRule;
 
 use crate::{block::check_merkle_root, filter::extract_relevant_info};
 
-#[derive(Debug)]
-pub struct BlockspaceProofInput {
-    pub block: Block,
-    pub rollup_params: RollupParams,
-    // TODO: add hintings and other necessary params
-}
-
 #[derive(Debug, BorshSerialize, BorshDeserialize)]
 pub struct BlockspaceProofOutput {
     pub header_raw: Vec<u8>,
     pub deposits: Vec<DepositInfo>,
     pub prev_checkpoint: Option<BatchCheckpoint>,
-    pub rollup_params_commitment: Buf32,
+    pub tx_filters_commitment: Buf32,
+    pub cred_rule: CredRule,
 }
 
-pub fn process_blockspace_proof(input: &BlockspaceProofInput) -> BlockspaceProofOutput {
-    let BlockspaceProofInput {
-        block,
-        rollup_params,
-    } = input;
-    assert!(check_merkle_root(block));
+pub fn process_blockspace_proof(
+    serialized_block: &[u8],
+    cred_rule: &CredRule,
+    serialized_tx_filters: &[u8],
+) -> BlockspaceProofOutput {
+    let block: Block = bitcoin::consensus::deserialize(serialized_block).unwrap();
+    assert!(check_merkle_root(&block));
     // assert!(check_witness_commitment(block));
```

Comment:
For `assert!(check_witness_commitment(block));` can this be removed or its something we have disabled for now. In that case we can leave a comment there.

Reply:
leave a FIXME TODO etc.